### DISCUSSION TITLE
Can we use filter and include query param on the same request?

### DISCUSSION THREAD

**Discussion by miazocool** [django-json-api/django-rest-framework-json-api#1139] [General]
Hello, thank u so much for the help.

Does the package have this feature; using sparse fields and include at the same time.

From the below url, i am trying to filter the "name" field from  the "device" resource, and at the same time "include" the related resource; "lots"
```
http://localhost:8000/api/v1/devices/UiouzK87qNxE?include=lots&fields[Device]=name
```
From json:api documentation, this feature is present, but not sure if it is present in the django json api package.
From json:api documentaiton :
```
GET /articles?include=author&fields[articles]=title,body HTTP/1.1
Accept: application/vnd.api+json
```

Here is the response that i get:
```json
{
    "data": {
        "type": "Device",
        "id": "UiouzK87qNxE",
        "attributes": {
            "name": "Scalpel"
        },
        "links": {
            "self": "http://localhost:8000/api/v1/devices/UiouzK87qNxE"
        },
        "meta": {
            "created_at": null,
            "created_by": null,
            "updated_at": null,
            "updated_by": null
        }
    }
}
```
The field filtering is correct, but the include related resource is not present. 

Thank u so much

**Comment by sliverc** [comment]
I assume you refer to the following [part of the spec](https://jsonapi.org/format/1.1/#fetching-sparse-fieldsets) where it states:

> Note: Because [compound documents](https://jsonapi.org/format/1.1/#document-compound-documents) require full linkage (except when relationship linkage is excluded by sparse fieldsets), intermediate resources in a multi-part path must be returned along with the leaf nodes. For example, a response to a request for comments.author should include comments as well as the author of each of those comments.

When I quickly looked at our implementation, it seems that fields which are ignored (not defined in sparse field set) also do not get included when requested through `include` parameter. Just ignoring the parameter as we currently do is certainly not compliant with the JSON:API spec but as a server MAY support an include parameter there are two options to adjust DJA.

1. Include resources even though there are not included in sparse fieldset (needs rewriting of the renderer)
2. raise an error if an include parameter is defined which is not defined in sparse field set (simpler to implement)

Option 1 is fairly involved. Could you therefore more elaborate on your use case why you need included resources even though you do not know by the request to what resource those are connected?

**Comment by miazocool** [comment]
Hello @sliverc ,

Actually, it was something wrong with my implementation,
I should include "lots" in the sparse fieldset.

By calling this:
devices/UiouzK87qNxE?include=lots&fields[Device]=ref,name,lots

I get the correct data. thank u
